Code/GraphMol/FileParsers/MolFileParser.cpp

```diff
@@ -932,6 +933,90 @@ void ParseAttachPointLine(RWMol *mol, const std::string &text,
   }
 }
 
+// the format differs between V2000 and V3000, so we have to do a bit of
+// translation here
+void ParseLinkNodeLine(RWMol *mol, const std::string &text, unsigned int line) {
+  PRECONDITION(mol, "bad mol");
+  PRECONDITION(text.substr(0, 6) == std::string("M  LIN"), "bad LIN line");
+
+  unsigned int nent;
+  try {
+    nent = FileParserUtils::stripSpacesAndCast<unsigned int>(text.substr(6, 3));
+  } catch (boost::bad_lexical_cast &) {
+    std::ostringstream errout;
+    errout << "Cannot convert '" << text.substr(6, 3) << "' to int on line "
+           << line;
+    throw FileParseException(errout.str());
+  }
+  std::string propVal = "";
+  unsigned int spos = 9;
+  for (unsigned int ie = 0; ie < nent; ie++) {
+    try {
+      auto aid = FileParserUtils::stripSpacesAndCast<unsigned int>(
+          text.substr(spos, 4));
+      if (!aid || aid > mol->getNumAtoms()) {
+        std::ostringstream errout;
+        errout << "LIN specification has bad atom idx on line " << line;
+        throw FileParseException(errout.str());
+      }
+      spos += 4;
+
+      if (text.size() < spos + 4 || text.substr(spos, 4) == "    ") {
+        std::ostringstream errout;
+        errout << "LIN specification missing repeat count on line " << line;
+        throw FileParseException(errout.str());
+      }
+      auto repeatCount = FileParserUtils::stripSpacesAndCast<unsigned int>(
+          text.substr(spos, 4));
+      spos += 4;
+      if (repeatCount < 2) {
+        std::ostringstream errout;
+        errout << "LIN specification: repeat count must be >=2 on line "
+               << line;
+        throw FileParseException(errout.str());
+      }
+      unsigned int substB = 0;
+      unsigned int substC = 0;
+      if (text.size() >= spos + 4 && text.substr(spos, 4) != "    ") {
+        substB = FileParserUtils::stripSpacesAndCast<unsigned int>(
+            text.substr(spos, 4));
+      }
+      spos += 4;
+      if (text.size() >= spos + 4 && text.substr(spos, 4) != "    ") {
+        substC = FileParserUtils::stripSpacesAndCast<unsigned int>(
+            text.substr(spos, 4));
+      }
+      spos += 4;
+
+      if (!substB || substB > mol->getNumAtoms() ||
+          substC > mol->getNumAtoms()) {
+        std::ostringstream errout;
+        errout << "LIN specification has bad substituent idx on line " << line;
+        throw FileParseException(errout.str());
+      }
+
+      boost::format formatter;
+      if (substC) {
+        formatter = boost::format("1 %1% 2 %2% %3% %2% %4%") % repeatCount %
```

Comment:
It seems somehow wasteful to not save these as unsigned int vects.
perhaps we could also save them as a vector or strings.

Reply:
In principle I agree, but I think it'd make more sense to have an actual data structure for `LINKNODE`s where we store the values in a structured way.
Once we've got this bugfix in so that the information no longer gets completely lost, I will do a PR (as an enhancement) that actually parses the `LINKNODE` information and sets up `StructureGroup` for it